db/migrate/20211110220355_set_new_collection_associations.rb

```diff
@@ -0,0 +1,44 @@
+# since we have changed the basic association types, we can no longer use the dynamic methods of :branding_group_id or
+# :user_id to reference back to the parent models.  Since the data exists in the documents, we can call :attributes to
+# load existing data and create maps with which to set new associations
+class SetNewCollectionAssociations < Mongoid::Migration
+  def self.up
+    # create maps showing existing associations
+    study_map = Study.all.map do |s|
```

Comment:
This is going to load all studies into memory in prod, which should be ok, but just to be safe, I might rewrite (also more concisely) as
```suggestion
    study_map = ActiveRecordUtils.pluck_to_hash(Study, [:id, :branding_group_id])
    study_map.each do |study_hash|
       if study_hash[:branding_group_id].present? && study_hash[:branding_group_id].is_a?(BSON::ObjectId)
         Study.find(study_hash[:id]).branding_groups.push(BrandingGroup.find(study_hash[:branding_group_id])
     end
```

and then something similar for the users.  This will have a few more queries to the DB (two per study-branding group pair), but much easier to read, and given that we only have a couple hundred or so studies in branding groups, that's much better than loading all studies into memory

Reply:
FWIW I tested out all this code (minus the assignments) in production to see how well it performed.  I'm confident we won't have issues with hanging.  But I take the point.

Reply:
e6441de